nameko_grpc/client.py

```diff
@@ -138,11 +142,25 @@ def default_compression(self):
         return "identity"
 
     def start(self):
-        self.channel = ClientChannel(self.target, self.ssl, self.spawn_thread)
-        self.channel.start()
+        if not self.lazy:
+            self._channel = self._get_channel()
+
+    def _get_channel(self):
+        with self._channel_creation_lock:
```

Comment:
This is in wrong place, or there should be another self._channel is None check after the lock. Other threads at the moment would wait but still create additional channels.

Reply:
You're right. 

I think it should be fixed now